src/ui/wxWidgets/MenuEditHandlers.cpp

```diff
@@ -53,28 +53,37 @@
 
 using pws_os::CUUID;
 
-//This function intentionally takes the argument by value and not by
-//reference to avoid touching an item invalidated by idle timeout
+/**
+ * This function intentionally takes the argument by value and not by
+ * reference to avoid touching an item invalidated by idle timeout.
+ */
 void PasswordSafeFrame::DoEdit(CItemData item)
 {
-  int rc = 0;
+  int returnCode = 0;
+
   if (!item.IsShortcut()) {
-    bool read_only = m_core.IsReadOnly() || item.IsProtected();
-    AddEditPropSheetDlg editDbox(this, m_core,
-                              read_only ? AddEditPropSheetDlg::SheetType::VIEW : AddEditPropSheetDlg::SheetType::EDIT,
-                              &item);
-    rc = editDbox.ShowModal();
-  } else {
-    EditShortcutDlg editDbox(this, m_core, &item);
-    rc = editDbox.ShowModal();
+    bool isItemReadOnly = m_core.IsReadOnly() || item.IsProtected();
+
+    AddEditPropSheetDlg dialog(
+      this, m_core,
+      isItemReadOnly ? AddEditPropSheetDlg::SheetType::VIEW : AddEditPropSheetDlg::SheetType::EDIT,
+      &item
+    );
+
+    returnCode = dialog.ShowModal();
   }
-  if (rc == wxID_OK) {
+  else {
+    EditShortcutDlg dialog(this, m_core, &item);
```

Comment:
EditShortcutDlg should also have r/w and r-o variants.

Reply:
I did only some code formatting and haven't had a look at the functionality of the shortcut editing dialog. I will do this later on a separate branch. Currently, in R-O mode the dialog doesn't allow modifications to a shortcut.